substra/sdk/compute_plan.py

```diff
@@ -101,8 +101,9 @@ def auto_batching(spec, is_creation: bool = True, batch_size: int = 20):
     testtuples = dict()
     if spec.testtuples:
         for testtuple in spec.testtuples:
+            # Rank 0 if testtuple.traintuple_id is in the nodes to ignore
             visited["test_" + testtuple.traintuple_id] = (
-                visited[testtuple.traintuple_id] + 1
+                visited.get(testtuple.traintuple_id, -1) + 1
```

Comment:
This is clever but it could be hard to maintain in the future, maybe just write an if-else clause ?